### DISCUSSION TITLE
Windows App SDK 1.6.0 (1.6.240829007)

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/WindowsAppSDK#4695] [Announcements]
## Windows App SDK 1.6.0 (1.6.240829007)

WinAppSDK 1.6.0 is the latest stable version of WinAppSDK providing new features and improvements to the platform.

**New features include:**
- Native AOT support
- Decoupled WebView2 versioning
- New Package Deployment APIs
- Improved TabView tear-out

**Additional updates include:**
- New wrapping mode for PipsPager
- More customization options for RatingControl

**To see everything that's new and changed, see the full [Windows App SDK 1.6 release notes](https://aka.ms/windowsappsdk/1.6/1.6.0/release-notes).**  

**See the [WinAppSDK 1.6 milestone](https://github.com/microsoft/WindowsAppSDK/milestone/17?closed=1) and [WinUI 3 in WinAppSDK 1.6 milestone](https://github.com/microsoft/microsoft-ui-xaml/milestone/21?closed=1) for a list of issues addressed in this release.**

### Try it out
- Download the [1.6.240829007 NuGet](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.6.240829007) package to use WinAppSDK 1.6 in your app.
- Download and update the [WinUI Gallery](https://www.microsoft.com/store/productId/9P3JFPWWDZRC) to see the WinUI 3 updates firsthand.

### Getting Started 

To get started using Windows App SDK to develop Windows apps, check out the following documentation:    
- [Install developer tools ](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment).     
- [Create a WinUI 3 app](https://docs.microsoft.com/windows/apps/winui/winui3/create-your-first-winui3-app).   
- [Continue your development journey](https://docs.microsoft.com/windows/apps/develop).   

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.6.0'>Windows App SDK 1.6.0 (1.6.240829007)</a>.</em>

**Comment by ynkdir** [comment]
Is Microsoft.WindowsAppRuntime.Bootstrap.dll removed from installer?

**Reply by DarranRowe** [reply]
It has never been part of the installer. It is in the Windows App SDK NuGet package under runtimes\win(10)-&lt;Platform&gt;\native.

![Screenshot 2024-09-05 182203](https://github.com/user-attachments/assets/51914c79-d89c-4949-a19b-76af3bbc7107)

It should also always be copied to the application directory or package by the build system.

**Reply by ThreeScreenStudios** [reply]
@ynkdir I had to delete the /obj and /bin folders, then rebuild and it didn't throw an error about Microsoft.WindowsAppRuntime.Bootstrap.dll anymore.

**Reply by ynkdir** [reply]
Thank you for reply.

I was using system installed bootstrap dll in "C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.*" by LoadLibrary so that I can distribute my library without bundling the dll.
It seems like I have to bundle bootstrap dll.

**Reply by DarranRowe** [reply]
That library isn't the a "system installed bootstrap dll". That is one of the core Windows App SDK libraries, and it is still part of the package.

![Screenshot 2024-09-06 140706](https://github.com/user-attachments/assets/af0150d0-9507-4ebe-a8a0-9f551af338e2)

But as stated, it is not where the exports for the bootstrap API are.

![Screenshot 2024-09-06 140836](https://github.com/user-attachments/assets/96c3478c-7994-4f1c-8109-3f45abbae6f7)

This is the list of exports from that library, and while the Windows App Runtime dynamic dependency API is exported from it, the bootstrapper isn't. I would have to double check the exports, but I don't think the bootstrapper API has ever been exported from it. The important thing is that both [MddBootstrapInitialize](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/win32/mddbootstrap/nf-mddbootstrap-mddbootstrapinitialize) and [MddBootstrapInitialize2](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/win32/mddbootstrap/nf-mddbootstrap-mddbootstrapinitialize2) are both documented to be exported from Microsoft.WindowsAppRuntime.Bootstrap.dll. This unfortunately means that if the exports were previously in Microsoft.WindowsAppRuntime.dll, it was an undocumented implementation detail and this can change.

**Reply by ynkdir** [reply]
What I wanted to say is that WindowsAppSDK 1.5 runtime package has Microsoft.WindowsAppRuntime.Bootstrap.dll in the installed directory.

"C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.5_5001.214.1843.0_x64__8wekyb3d8bb\Microsoft.WindowsAppRuntime.Bootstrap.dll"

And I can call MddBootstrapInitialize() in the dll.

But WindowsAppSDK 1.6 runtime package doesn't have Microsoft.WindowsAppRuntime.Bootstrap.dll in the installed directory.

I don't complain to it.  I just wanted to confirm that this is intented.

**Reply by DarranRowe** [reply]
Well, there is documentation hinting that Microsoft.WindowsAppRuntime.Bootstrap.dll being part of the Windows App Runtime package is a mistake.

In the "[Use the Windows App SDK runtime for apps packaged with external location or unpackaged](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/use-windows-app-sdk-run-time#using-the-bootstrapper-api)" documentation, the section named "Using the bootstrapper API" states:

"That library is a small DLL that you must distribute with your app; it's not part of the framework package itself."

So the fact that it was found as part of the framework package is, by this documentation, a mistake.

**Reply by ynkdir** [reply]
I understand.  Thank you.

**Comment by Slion** [comment]
When updating a brand new project from Template Studio from 1.4 to 1.6 it wants net8 which is using the new RuntimeIndentifiers like `win-x64` instead of ̀ win10-x64` but even after fixing those the template Core DLL would still not compile complaining about RID. Not sure if that's because of dependency or something. I'll stick to net7 and v1.5 for now.

**Reply by ghost1372** [reply]
did you change rid in pubxml files?

**Reply by Slion** [reply]
Not sure what those are, so probably not 😁 Ok found them… I'll give it a try then. Should have done a search for `win10-` really 🫣

**Reply by Slion** [reply]
Ok so I could go one step further but then it seems quite a few NuGet dependencies are not yet available for net8 so I'll definitely stick to v1.5 for now.

**Comment by Diegorro98** [comment]
In the release notes, at `New APIs`, the following new APIs are mentioned:

> ```cs
> Microsoft.Windows.Storage
> 
>     ApplicationData
>     ApplicationDataContainer
>     ApplicationDataContract
>     ApplicationDataCreateDisposition
>     ApplicationDataLocality
> ```

I have been using `Windows.Storage` API for all my WASDK projects and I would like to know which are the differences or the reasons to bring this new API, because if it has advantages, it might be interesting to use the new one. Or maybe it is just a renaming...

**Reply by ghost1372** [reply]
There are some new API that can be used in unpackaged app but not released yet
https://github.com/microsoft/WindowsAppSDK/pull/4057

Here you can see its changes and benefits

https://github.com/microsoft/WindowsAppSDK/blob/main/specs/applicationdata/ApplicationData.md

**Comment by duwenlong2** [comment]
I found during my search that App sdk 1.6 seems to support AOT compilation for WinUI3. Since I couldn't find the sample program, I tried following the documentation process but didn't experience AOT compilation of WinUI3. Can you provide an example of this? Or can you give me some help? A C # empty WinUI3 project was created with the following configuration:
![image](https://github.com/user-attachments/assets/9210b5b3-4bcc-4d4b-b6d8-f80b8c102fd6)

Refer to the following two documents:
https://learn.microsoft.com/zh-cn/windows/apps/windows-app-sdk/stable-channel#version-16

https://learn.microsoft.com/zh-cn/dotnet/core/deploying/native-aot/?tabs=windows%2Cnet9plus#publish-native-aot-using-the-cli

**Reply by manodasanW** [reply]
Did you create a MSIX package for your app?  The publishing step is when AOT compilation happens.
